### ISSUE TITLE
lcd.isVisible() has a memory problem

### ISSUE THREAD

**Issue by Storchy1** [FrSkyRC/ETHOS-Feedback-Community#4951]
XE with Ethos 1.6.0.
I have made two small scripts to prove the MemLeak of lcd.isVisible()
Installation:
Make a new model.
The model has the script ShowRam on screen 1, it shows the current availableRam and the sec since the remote control XE is on.
on screen 2 the script ShowDateTime is loaded. This script shows the date & time of the XE RC and whether lcd.isVisible() is used or not.You must go to Configure for screen 2  when configuring ShowDateTime and set use isVisible() to true / false.
If the use of isVisible is true i lose about 900kByte of memory in ~10 minutes, if the script ShowRam is the active script.
Then i go to configure and change use isVisible() to false and i lose no more memory.

Please check the function lcd.isVisible()
[ShowRam.zip](https://github.com/user-attachments/files/18282616/ShowRam.zip)

**Comment by uj-berlin** [comment]
I have the same problem with my XE running Ethos 1.5.19
Only a small script for showing switch positions.
Lose of memory when lcd.isVisible() is used and the widget is in the backgrond. No problem in foreground.
No problem at all withtout isVisible() in the script.

**Comment by bsongis-frsky** [comment]
I doubt the problem is in `lcd.isVisible`, it is a very simple function returning a bool

**Comment by uj-berlin** [comment]
I have tested this with a short script. The only difference was the query isVisible(), nothing else. Without the query no ram leak, with the query and the script in the background lose of memory until a warning after about 30 minutes.

**Comment by bsongis** [comment]
What happens if your `paint` method is empty (no string creation)?

**Comment by Storchy1** [comment]
**_local function paint(widget)
end_**
I have change the paint method, it is complete empty,  and the ram leak is still there!
I can't understand why this should be important, since the script with the empty paint method is always in the background during this test. It is never called.

**Comment by bsongis** [comment]
May I assume that no other script is running?

**Comment by Storchy1** [comment]
Off course, no other script is running.
it is a empty model only with this two scripts in the zip File with was added to my first comment
Just try it and you will see that the ram gets down.
I have not only testet by myself, others in the german Engel forum testet the scripts and they have the same behavoir

**Comment by bsongis-frsky** [comment]
I am doing the test now, on X20-PRO

On the first screen I have:
 - RAM available
 - ShowRAM
 - ShowDateTime with isVisible ON in Configuration

On the second screen I have:
- ShowDateTime with isVisible ON in Configuration

Waiting the 10 minutes as requested by the issue report

**Comment by bsongis-frsky** [comment]
Did you wait 10 minutes without doing anything? Perhaps you have looked at your bitmaps and encounter this other issue?
#4943

RAM doesn't move here. I will have a coffee.

**Comment by Storchy1** [comment]
Screen 1  must be the active one and i do nothing.
if screen 2 is the active one no ram issue.
I look on the ram value in the midlle of the screen and i goes down.
enjoy your coffee

**Comment by bsongis-frsky** [comment]
You are right. A coffee costs 2MB

**Comment by Storchy1** [comment]
Ok thanks for the info, I am glad that it also leads to a loss of RAM for you. Now it is possible to find the error

**Comment by bsongis-frsky** [comment]
And now the script is killed!

**Comment by bsongis-frsky** [comment]
Do you know which script uses the RAM? Because you have given 2 scripts ...

**Comment by bsongis-frsky** [comment]
Why your method wakeup returns `true`?
It should return nothing

**Comment by Storchy1** [comment]
No, but if you configure the script **ShowDateAndTime** useIsVisible to off there are no RAM lost anymore.
So we thought it must have something to do with this function. Both scripts have no other function that can use RAM

**Comment by bsongis-frsky** [comment]
Testing this script now. It **gives** the same memory leak

```lua
local function wakeup(widget)
  return true
end

local function paint(widget)
end

local function init()
    system.registerWidget({key="Test", name="Memory Leak", wakeup=wakeup, paint=paint})
end

return {init=init}
```

**Comment by Storchy1** [comment]
Oh, i did not know, that it is forbidden to return true in the wakeup routine
Thanks for the info

**Comment by Storchy1** [comment]
I will test here the scripts without return true

**Comment by Storchy1** [comment]
I testet the script without return true, i used only return . No more RAM leak! I am happy!
That is a lit bit strange for me, that a aditional return parameter will cause a Ram leak.
But it doesn't matter to me i will have a look on all my wakeup handler and kill the true rarameter after return.
Thank you
